### ISSUE TITLE
[Bug]: BrazeUICompat refuses to compile with some configurations

### ISSUE THREAD

**Issue by alansteiman** [braze-inc/braze-swift-sdk#32]
### Platform

iOS

### Platform Version

iOS 15.2

### Braze SDK Version

5.8.0

### Xcode Version

13.4.1

### Computer Processor

Intel

### Repro Rate

100%

### Steps To Reproduce

1. Podfile looking like this:
  ```
  $brazeVersion = '5.8.0'
def brazePod
    pod 'BrazeKitCompat', $brazeVersion
    pod 'BrazeUICompat', $brazeVersion
end
```
2. Compile
  

### Expected Behavior

Compile correctly

### Actual Incorrect Behavior

Getting the following errors while compiling:
  
Braze SDK:
<img width="823" alt="Screen Shot 2023-01-05 at 1 18 57 PM" src="https://user-images.githubusercontent.com/90182796/210744789-aedde4b1-b6a3-4166-96de-29583ded3fe0.png">

My project:
<img width="774" alt="Screen Shot 2023-01-05 at 1 19 12 PM" src="https://user-images.githubusercontent.com/90182796/210744834-78400cd3-452b-478e-adf0-c3730e5d1421.png">

  

### Additional Information

Already tried this: https://developer.apple.com/forums/thread/128535

**Comment by hokstuff** [comment]
Hi @alansteiman,

Are you following the [Swift SDK migration guide steps](https://braze-inc.github.io/braze-swift-sdk/documentation/braze/appboy-migration-guide/)? If you want to get your app up and running with the new SDK ASAP, you can follow Scenario 1 and be sure to update your imports as well as make changes around the specific dependency manager in your project. Also, are you using any relevant custom build flags in your project?

**Comment by alansteiman** [comment]
@hokstuff thanks for the quick response.

Yes I am following the scenario 1 of that migration guide. This is how I did:

1) Replace `Appboy-iOS-SDK` by `BrazeKitCompat` in the podfile and run pod install
2) Replace all the imports in both Swift and Objective-C
3) Compile error, the class `ABKInAppMessageSlideupViewController` was not found
4) Added `BrazeUICompat` in the podfile, same version as `BrazeKitCompat` (5.8.0) and run pod install
5) Compile error: `Include of non-modular header inside framework module`

<img width="777" alt="Screen Shot 2023-01-06 at 11 30 07 AM" src="https://user-images.githubusercontent.com/90182796/210952132-396e7fcc-15cc-4f9c-9c71-47716abd1e11.png">

**Comment by hokstuff** [comment]
I see, it looks like you're running into linking issues during your Cocoapods migration. 

[This article](https://stackoverflow.com/questions/27776497/include-of-non-modular-header-inside-framework-module) suggests some workarounds for that error as well as possible root causes based on which headers are linked. I also have a few questions:
- Is your host app written in Swift, ObjC, or a mix?
- Are you doing any special integration, or are you using a vanilla Cocoapods integration?
- Do you have any custom linker flags in your project?
- Have you tried building on Xcode 14?

**Comment by alansteiman** [comment]
Hi @hokstuff, answering your questions:

- It's a mix
- Definitely not vanilla Cocoapods
- Haven't identified any custom linker flag
- Same problem on iOS 14

I tried integrating the Braze pods in a new project and works fine, definitely it's something related to our massive Podfile.
I will close this issue for now and continue investigating the root cause, whenever I find it will post it here for history purpose.

**Comment by LucioPham** [comment]
Hi @alansteiman @hokstuff 

I'm also facing this issue. My project was also set up with a mix between Swift and ObjC.

I tried to build on the latest XCode 14 but still errors.

@alansteiman When you trying on a new project, did you try with a mix between Swift and ObjC?

**Comment by alansteiman** [comment]
> Hi @alansteiman @hokstuff
> 
> I'm also facing this issue. My project was also set up with a mix between Swift and ObjC.
> 
> I tried to build on the latest XCode 14 but still errors.
> 
> @alansteiman When you trying on a new project, did you try with a mix between Swift and ObjC?
> 
> I was also facing this issue, as seen in the image below.
> 
> <img alt="Screenshot 2023-01-17 at 11 38 57" width="292" src="https://user-images.githubusercontent.com/115771843/212811259-888eef04-528d-40bb-9b11-7220edd2b9c0.png">

Hi @LucioPham, on a new project was only Swift

**Comment by LucioPham** [comment]
> I see, it looks like you're running into linking issues during your Cocoapods migration.
> 
> [This article](https://stackoverflow.com/questions/27776497/include-of-non-modular-header-inside-framework-module) suggests some workarounds for that error as well as possible root causes based on which headers are linked. I also have a few questions:
> 
> * Is your host app written in Swift, ObjC, or a mix?
> * Are you doing any special integration, or are you using a vanilla Cocoapods integration?
> * Do you have any custom linker flags in your project?
> * Have you tried building on Xcode 14?

Hi @alansteiman, I believe this issue happened on a mixed project. 

@hokstuff Could you help to take a look? I did try all workarounds from the article you mentioned but none of them work.

**Comment by LucioPham** [comment]
Hi @alansteiman,sorry for bothering you but could we have a private conversation for discuss about this issue ?

**Comment by lowip** [comment]
@alansteiman, @LucioPham 

I'm re-opening this issue as we've been able to reproduce this exact error message with a different setup. 
We have identified a fix, and we'll update here once it's released.

**Comment by LucioPham** [comment]
Hi @lowip, thank you for your support, I also found other issues after integrating with the new Braze SDK. Should I create a new ticket issue for them?

**Comment by lowip** [comment]
Hi @LucioPham, 

Please open new tickets for other issues.

Thanks

**Comment by LucioPham** [comment]
Hi @alansteiman, could you share your Podfile ?

**Comment by alansteiman** [comment]
> Hi @alansteiman, could you share your Podfile ?

The podfile I have is highly attached to my company, will try to reproduce the issue at the smallest possible scale

**Comment by LucioPham** [comment]
Hi @alansteiman, @lowip , It really weird when I remove the line `inhibit_all_warnings!` in my Podfile the error just gone and I be able build successfully.

**Comment by alansteiman** [comment]
> 

I don't have that line at all

**Comment by alansteiman** [comment]
> Hi @alansteiman, @lowip , It really weird when I remove the line `inhibit_all_warnings!` in my Podfile the error just gone and I be able build successfully.

I don't have that line at all

**Comment by hokstuff** [comment]
Hi @alansteiman @LucioPham,

We have released Swift SDK [version 5.9.0](https://github.com/braze-inc/braze-swift-sdk/releases/tag/5.9.0) which addresses this issue. Please let us know if you run into further problems when using that version. Thanks!

**Comment by alansteiman** [comment]
Thanks @hokstuff, I tried the version and was able to compile, archive and distribute via Testflight (using Xcode 14+), appreciate the effort 🙌
